http-server-netty/src/test/groovy/io/micronaut/http/util/CookieUtilSpec.groovy

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.util
+
+import io.micronaut.http.cookie.Cookie
+import io.micronaut.http.cookie.SameSite
+import spock.lang.Specification
+
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.Month
+import java.time.temporal.ChronoUnit
+
+class CookieUtilSpec extends Specification {
+
+    static long durationInSeconds(Integer year, Month month, Integer day, Integer hour, Integer minute, Integer second){
+        return Duration.between(LocalDateTime.of(year, month, day, hour, minute, second), LocalDateTime.now())
+                .get(ChronoUnit.SECONDS)
+    }
+
+    void "test parsing cookie date from header #setCookieHeader"() {
+        given:
+        Cookie cookie = CookieUtil.getCookieFromString("$setCookieHeader")
```

Comment:
since you are not creating something new with `setCookieHeader` there is no need to create a new GString from `setCookieHeader`.

```suggestion
        Cookie cookie = CookieUtil.getCookieFromString(setCookieHeader)
```